### DISCUSSION TITLE
non-interleaved AVI

### DISCUSSION THREAD

**Discussion by Marek68** [baSSiLL/SharpAvi#58] [Q&A] [answered]
Hello,

Firstly I quickly describe my workflow:
My service is receiving raw RGBA frames from rendering engine. Received frames are written via SharpAvi library into avi stream. Created AVI file should overlay other mxf (background) file via ffmpeg. 
Problem is that created AVI file (via SharpAvi) is non-interleaved and it is not visible after overlay (after overlay is visible only background mxf video). Ffmpeg command is correct, I tried it with other AVI and it works.
Created AVI does not contain any audio tracks, may it cause problem with non interleaved?

**Comment by baSSiLL** [comment, marked as answer]
Hi
Why do you talk about interleaved? Do you get some complain from FFMPEG about this? Apparently an AVI cannot be interleaved if it only has a single stream. But you don't need anything but video as I get.
Do you correctly set alpha values for pixels?
The other AVI that worked with FFMPEG, did it contain the alpha channel?

**Reply by Marek68** [reply]
Hi, 
I am sorry for my previous question. I already solved my problem. As I wrote, in my code is needed to reorganize colors order from RGBA to BGRA and I forgot to copy value for alpha channel. Based on your questions I check my code and found my fail. 
Now everything works fine.

Thanks a lot!
